src/App/Fossa/Config/Common.hs

```diff
@@ -209,8 +213,25 @@ metadataOpts =
 releaseGroupMetadataOpts :: Parser ReleaseGroupMetadata
 releaseGroupMetadataOpts =
   ReleaseGroupMetadata
-    <$> strOption (applyFossaStyle <> long "release-group-name" <> stringToHelpDoc "The name of the release group to add this project to")
-    <*> strOption (applyFossaStyle <> long "release-group-release" <> stringToHelpDoc "The release of the release group to add this project to")
+    <$> strOption (applyFossaStyle <> long "release-group-name" <> stringToHelpDoc "The name of the release group to add this project to" <> internal)
+    <*> strOption (applyFossaStyle <> long "release-group-release" <> stringToHelpDoc "The release of the release group to add this project to" <> internal)
+
+deprecateReleaseGroupMetadata :: Has Diagnostics sig m => ProjectMetadata -> m ProjectMetadata
+deprecateReleaseGroupMetadata projectMetadata = do
+  case (projectReleaseGroup projectMetadata) of
+    Nothing -> pure projectMetadata
+    Just _ -> do
+      warn deprecationMessage
+      pure $ removeReleaseGroupMetadata projectMetadata
+  where
+    removeReleaseGroupMetadata :: ProjectMetadata -> ProjectMetadata
+    removeReleaseGroupMetadata project = project{projectReleaseGroup = Nothing}
+
+    deprecationMessage :: Text
+    deprecationMessage =
+      renderIt $
+        vsep
+          [annotate (color Red) "Release group options for this command have been deprecated. Refer to `fossa release-group` subcommands to interact with FOSSA release groups."]
```

Comment:
Thinking more about this, not sure if we want to deprecate this on analyze. There was a customer that was asking to update the release projects upon scan and if we deprecate this, they wouldn't be able to do this.

Reply:
That's true, you would be able to run `analyze` and then update the release projects with `release-group add-projects`, but I think the customer was looking to only run `fossa analyze` to have their release group project updated.

[slack thread](https://teamfossa.slack.com/archives/C043EM3L96Z/p1710800449974419?thread_ts=1710450155.589159&cid=C043EM3L96Z)

Reply:
It sounds like deprecating them would not totally prevent the customer from updating a project in a release group. I think it's OK in that case to say that they'll need to use these new commands even if they have a preference for `analyze`.

Reply:
That sounds good to me